### ISSUE TITLE
Don't throw unused error for _ in a destructured array

### ISSUE THREAD

**Issue by cevek** [Microsoft/TypeScript#25409]
**TypeScript Version:**  3.0.0-dev.20180626

**Search Terms:**  Unused variable

**Code**

```ts
const [_, a] = [1, 2];
for (const [_, value] of map) {}
```
for function arguments works as expected
```ts
function foo(_: number) {} // no error
```

**Expected behavior:**
No error

**Actual behavior:**
Error

**Comment by ajafff** [comment]
IMO this is not the same as the parameter example.
Parameters cannot be omitted, values in array destructuring can.

**Comment by RyanCavanaugh** [comment]
Hm, true. If this is intended why not write
```ts
const [ , a] = [1, 2];
```
?

**Comment by cevek** [comment]
Oh, yeah, didn't know about it. Close this issue
